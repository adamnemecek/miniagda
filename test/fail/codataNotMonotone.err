MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "codataNotMonotone.ma" ---
--- scope checking ---
--- type checking ---
type  SNat : + Size -> Set
term  zero : .[s!ze : Size] -> .[i < s!ze] -> SNat s!ze
term  zero : .[i : Size] -> < zero i : SNat $i >
term  succ : .[s!ze : Size] -> .[i < s!ze] -> ^ SNat i -> SNat s!ze
term  succ : .[i : Size] -> ^(y1 : SNat i) -> < succ i y1 : SNat $i >
type  NatEq : -(i : Size) -> ^ SNat i -> ^ SNat i -> Set
term  eqz : .[i : Size] -> < eqz i : NatEq $i (zero [i]) (zero [i]) >
error during typechecking:
NatEq
/// constructor eqs
/// szConstructor NatEq : .[i <= #] -> .[n : SNat i] -> .[m : SNat i] -> ^(y3 : NatEq i n m) -> < eqs i n m y3 : NatEq $i (succ [i] n) (succ [i] m) >
/// new i <= #
/// szSizeVarUsage of i in .[n : SNat i] -> .[m : SNat i] -> ^(y3 : NatEq i n m) -> < eqs i n m y3 : NatEq $i (succ [i] n) (succ [i] m) >
/// checking SNat i  to be antitone in variable i
/// leqVal' [(i,0)] |- SNat i  <=-  SNat $i : Set #
/// leqVal' [(i,0)] |- i  <=-  $i : <= #
/// leSize v0 <=- ($ v0)
/// leSize' ($ v0) <= v0
/// leSize: 0 + 1 <= 0 failed
