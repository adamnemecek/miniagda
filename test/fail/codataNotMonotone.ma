-- 2010-05-06

sized data SNat : Size -> Set 
{ zero : [i : Size] -> SNat ($ i)
; succ : [i : Size] -> SNat i -> SNat ($ i) 
}

sized codata NatEq : (i : Size) -> SNat i -> SNat i -> Set
{ eqz : [i : Size] -> NatEq ($ i) (zero i) (zero i)
; eqs : [i : Size] -> (n : SNat i) -> (m : SNat i) -> 
   NatEq i n m -> NatEq ($ i) (succ i n) (succ i m)
}
