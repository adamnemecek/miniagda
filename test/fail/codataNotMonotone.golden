MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "codataNotMonotone.ma" ---
--- scope checking ---
--- type checking ---
type  SNat : + Size -> Set
term  SNat.zero : .[s!ze : Size] -> .[i < s!ze] -> SNat s!ze
term  SNat.zero : .[i : Size] -> < SNat.zero i : SNat $i >
term  SNat.succ : .[s!ze : Size] -> .[i < s!ze] -> ^ SNat i -> SNat s!ze
term  SNat.succ : .[i : Size] -> ^(y1 : SNat i) -> < SNat.succ i y1 : SNat $i >
type  NatEq : -(i : Size) -> ^ SNat i -> ^ SNat i -> Set
term  NatEq.eqz : .[i : Size] -> < NatEq.eqz i : NatEq $i (SNat.zero [i]) (SNat.zero [i]) >
error during typechecking:
NatEq
/// constructor NatEq.eqs
/// szConstructor NatEq : .[i : Size] -> .[n : SNat i] -> .[m : SNat i] -> ^(y3 : NatEq i n m) -> < NatEq.eqs i n m y3 : NatEq $i (SNat.succ [i] n) (SNat.succ [i] m) >
/// new i <= #
/// szSizeVarUsage of i in .[n : SNat i] -> .[m : SNat i] -> ^(y3 : NatEq i n m) -> < NatEq.eqs i n m y3 : NatEq $i (SNat.succ [i] n) (SNat.succ [i] m) >
/// checking SNat i  to be antitone in variable i
/// leqVal'  SNat i  <=-  SNat $i : Set #
/// leqVal'  i  <=-  $i : Size
/// leSize i <=- $i
/// leSize' $i <= i
/// leSize: 0 + 1 <= 0 failed
